src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+
+import java.nio.file.Path;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+
+/**
+ * This class encapsulates a local database for Duke bot.
+ */
+public class Storage {
+    private final Path DATABASE_PATH;
+
+    public Storage(String filePath) {
+        this.DATABASE_PATH = java.nio.file.Paths.get(
+            System.getProperty("user.dir"), filePath);
+
+        if (!databaseExist()) {
+            createDatabase();
+        }
+    }
+
+    /**
+     * Creates a local text file for saving the lists of tasks if it does not exist.
+     */
+    public void createDatabase() {
+        File database = new File(DATABASE_PATH.toString());
+
+        try {
+            database.getParentFile().mkdirs();
+            database.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Checks if the database exists.
+     *
+     * @return True if the text file exists.
+     */
+    public boolean databaseExist() {
+        File database = new File(DATABASE_PATH.toString());
+        return database.exists();
+    }
+
+
+    /**
+     * Parses the string of tasks from the database into Task objects.
+     *
+     * @return The list of tasks stored in the database.
+     * @throws DukeException If corrupted database.
+     */
+    protected ArrayList<Task> readFromDatabase() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        try {
+            FileReader reader = new FileReader(DATABASE_PATH.toString());
+            BufferedReader bufferedReader = new BufferedReader(reader);
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                taskList.add(stringToTask(line));
+            }
+            reader.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return taskList;
+    }
+
+    /**
+     * Updates the database.
+     *
+     * @param taskList The list containing the tasks to be saved.
+     */
+    public void writeToDatabase(ArrayList<Task> taskList) {
```

Comment:
Methods for operations on the database are written as verbs, clear and easy to understand without having to read the JavaDoc comment.